src/api/config.c

```diff
@@ -1037,6 +1037,16 @@ int api_config(struct ftl_conn *api)
 	if(api->method == HTTP_GET)
 		return api_config_get(api);
 
+	// Check if this is an app session and reject the request if app sudo
+	// mode is disabled
+	if(api->session != NULL && api->session->app && !config.webserver.api.app_sudo.v.b)
+	{
+		return send_json_error(api, 403,
+		                       "forbidden",
+		                       "config read-only",
+		                       "app session but webserver.api.app_sudo is false");
```

Comment:
Maybe we should use a more detailed and user friendly message:

```suggestion
		                       "Unable to change configuration (read-only)",
		                       "The current app session is not allowed to modify Pi-hole config settings (webserver.api.app_sudo is false)");
```

Reply:
I agree with rd